BayeuxClient.agent.lib.nut

```diff
@@ -0,0 +1,936 @@
+        if (!(topic in _userHandlers)) {
+            onDone && onDone(Bayeux.Error(BAYEUX_CLIENT_ERROR_TYPE.LIBRARY_ERROR, BAYEUX_CLIENT_LIBRARY_ERROR_NOT_SUBSCRIBED));
+            return;
+        }
+
+        local msgId = _nextMessageId();
+        local message = {
+            "id" : msgId,
+            "clientId" : _clientId,
+            "channel" : BAYEUX_CLIENT_META_CHANNEL_UNSUBSCRIBE,
+            "subscription" : topic
+        }
+
+        local onResponse = function(response) {
+            _onUnsubscribeResponse(response, topic, onDone);
+        }.bindenv(this);
+
+        _metaHandlers[msgId] <- onResponse;
+
+        local sent = function(error) {
+            if (error != null) {
+                if (msgId in _metaHandlers) {
+                    delete _metaHandlers[msgId];                
+                }
+                onDone && onDone(error);
+            }
+        }.bindenv(this);
+
+        _enqueueMessage(message, sent);
+    }
+
+    // Sets onConnected callback.
+    //
+    // Parameters:
+    //     onConnected : Function       Callback called every time the client is connected.
+    //                                  The callback signature:
+    //                                  onConnected(error), where
+    //                                      error :             null if the connection is successful, error details otherwise.
+    //                                          Bayeux.Error
+    //
+    // Returns:                         Nothing.
+    function setOnConnected(callback) {
+        _onConnectedCb = callback;
+    }
+
+    // Sets onDisconnected callback.
+    //
+    // Parameters:
+    //     onDisconnected : Function    Callback called every time the client is disconnected.
+    //                                  The callback signature:
+    //                                  onDisconnected(reason), where
+    //                                      reason :            null if the disconnection was caused by the disconnect() method,
+    //                                          Bayeux.Error    error details which explains a reason of the disconnection otherwise.
+    //
+    // Returns:                         Nothing.
+    function setOnDisconnected(callback) {
+        _onDisconnectedCb = callback;
+    }
+
+    // Enables or disables the client debug output. Disabled by default.
+    //
+    // Parameters:
+    //     value : Boolean              true to enable, false to disable
+    //
+    // Returns:                         Nothing.
+    function setDebug(value) {
+        _debugEnabled = value;
+    }
+
+    // -------------------- PRIVATE METHODS -------------------- //
+
+    function _init() {
+        _cookies = {
+            // TODO: We should get this cookie from Bayeux server
+            "BAYEUX_BROWSER" : "BAYEUX_BROWSER=" + imp.configparams.deviceid
+        }
+
+        _advice = {
+            "timeout" : BAYEUX_CLIENT_DEFAULT_ADVICE_TIMEOUT * 1000,
```

Comment:
All the internal time should be in seconds

Reply:
Unless there is a really strong reason why to have millis...

Reply:
It's just a requirement of the protocol. We have to use millis in some places

Reply:
ok